test/core/Do.Test.Architecture/Architecture/Application/AddingExtensions.cs

```diff
@@ -100,4 +102,36 @@ public void Layers_are_skipped_when_they_provide_no_configuration_target()
 
         feature.VerifyConfiguresNothing();
     }
+
+    [Test]
+    public void Adding_the_same_layer_more_than_once_gives_error()
+    {
+        var forge = GiveMe.AForge();
+        var layer = MockMe.ALayer();
+
+        var forgeAction = () => forge.Application(app =>
+        {
+            app.Layers.Add(layer);
+            app.Layers.Add(layer);
+        });
+
+        forgeAction.ShouldThrow<InvalidOperationException>().Message.ShouldBe(
+            $"Cannot add `{layer.Id}`, it was already added.");
```

Comment:
here you need to set an id for the layer, and assert an hardcoded message. avoid using variables in assertions.